src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h

```diff
@@ -553,38 +577,60 @@ namespace OpenMS
           Size spec_a = consensus_spec.size(), spec_b = exp[*sit].size(), align_size = alignment.size();
           for (typename MapType::SpectrumType::ConstIterator pit = exp[*sit].begin(); pit != exp[*sit].end(); ++pit)
           {
-            // either add aligned peak height to existing peak
-            if (alignment.size() > 0 && alignment[align_index].second == spec_b_index)
-            {
-              consensus_spec[alignment[align_index].first].setIntensity(consensus_spec[alignment[align_index].first].getIntensity() +
-                                                                        pit->getIntensity());
-              ++align_index; // this aligned peak was explained, wait for next aligned peak ...
-              if (align_index == alignment.size()) alignment.clear();  // end reached -> avoid going into this block again
-            }
-            else // ... or add unaligned peak
-            {
-              consensus_spec.push_back(*pit);
-            }
-            ++spec_b_index;
+              if (alignment.size() == 0 || alignment[align_index].second != spec_b_index)
+                  // ... add unaligned peak
+              {
+                  consensus_spec.push_back(*pit);
+              }
+              // or add aligned peak height to ALL corresponding existing peaks
```

Comment:
I am amenable to changing this to divide the peak intensity across all of the matching peaks. I will make the changes and add them to this request.